.github/actions/publish/action.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: build and draft GitHub release
+# https://github.com/product-os/flowzone/tree/master/.github/actions
+inputs:
+  json:
+    description: "JSON stringified object containing all the inputs from the calling workflow"
+    required: true
+  secrets:
+    description: "JSON stringified object containing all the secrets from the calling workflow"
+    required: true
+  variables:
+    description: "JSON stringified object containing all the variables from the calling workflow"
+    required: true
+
+runs:
+  # https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        docker compose -f docker-compose.build.yml up build \
+          --exit-code-from build || { docker compose logs ; exit 1 ; }
```

Comment:
No need for `--exit-code-from` with only one service.

Also no need to catch and print the logs with only one service, since they will already be in the foreground.